### PR TITLE
Make single-use functions internal and static

### PR DESCRIPTION
**Pull request by PaulWessel**
Moving a bunch of functions (~20-30) from external to internal since they are only used within a single file and thus should not be exported.

### CODE REVIEW

**Comment by seisman** [comment]
A warning if `USE_COMMON_LONG_OPTIONS` is not defined.
```
../src/gmt_init.c:488:16: warning: unused function 'gmtinit_handle_escape_text' [-Wunused-function]
GMT_LOCAL void gmtinit_handle_escape_text (char *text, char key, int way) {
```

**Comment by seisman** [comment]
Another two warnings:
```
In file included from ../src/gmt_customio.c:1724:
../src/gmt_gdalwrite.c:177:17: warning: comparison of integers of different signs: 'uint64_t' (aka 'unsigned long long') and 'int' [-Wsign-compare]
                for (k = 0; k < n_colors; k++) {
                            ~ ^ ~~~~~~~~
../src/gmt_gdalwrite.c:185:19: warning: comparison of integers of different signs: 'uint64_t' (aka 'unsigned long long') and 'int' [-Wsign-compare]
                                for (k = 0; k < n_colors; k++)
                                            ~ ^ ~~~~~~~~
2 warnings generated.

```

**Comment by joa-quim** [comment]
DO NOT MERGE yet. Removing ``EXTERN_MSC `` may break externals. I have no time today to test it but I'm afraid of Julia breaks.

**Comment by PaulWessel** [comment]
We can wait for you to verity tomorrow.  But doubt you are calling these from Julia - you are just using the API calls.

**Comment by joa-quim** [comment]
I'll have time latter today. But I'm not calling only API functions. I can call any function that is exported. Changing that is a kind of breakage.

**Comment by PaulWessel** [comment]
It is, but none of the internals are guaranteed to be available to externals.  We can adjust of course but we don't want the externals to be so fragile that they depend on lower-level undocumented things.  This is what Caress did and suddenly we get complaints that some long-forgotten library function was used by him...

**Comment by joa-quim** [comment]
Nothing is detectable broken on Julia side.

It's not the same as Dave's case that was (is) accessing lower level functions for grid reading when there are API functions to do that, I need, for example, to access to ``gmt_manage_workflow`` or ``gmt_getpen`` or structs like ``GMT_DATASEGMENT``. Without these there is no Julia wrapper.

File: src/gmt_remote.c [hunk 1]
```diff
@@ -274,7 +274,7 @@ GMT_LOCAL int gmthash_get_url (struct GMT_CTRL *GMT, char *url, char *file, char
 	return 0;
 }
 
-struct GMT_DATA_HASH * hash_load (struct GMT_CTRL *GMT, char *file, int *n) {
+GMT_LOCAL struct GMT_DATA_HASH * hash_load (struct GMT_CTRL *GMT, char *file, int *n) {
```
Comment (joa-quim) [inline src/gmt_remote.c]: Aren't all GMT_xx public?

File: src/gmt_support.c [hunk 2]
```diff
@@ -7210,7 +7210,7 @@ int gmt_list_cpt (struct GMT_CTRL *GMT, char option) {
 }
 
 /*! . */
-void gmtlib_make_continuous_colorlist (struct GMT_CTRL *GMT, struct GMT_PALETTE *P) {
+GMT_LOCAL void support_make_continuous_colorlist (struct GMT_CTRL *GMT, struct GMT_PALETTE *P) {
```
Comment (joa-quim) [inline src/gmt_support.c]: Misses the gmt prefix? (gmt)support_make_continuous_colorlist

File: src/gmt_support.c [hunk 3]
```diff
@@ -13858,7 +13858,7 @@ unsigned int gmtlib_pow_array (struct GMT_CTRL *GMT, double min, double max, dou
 }
 
 /*! . */
-uint64_t gmt_time_array (struct GMT_CTRL *GMT, double min, double max, double inc, char unit, bool interval, double **array) {
+GMT_LOCAL uint64_t support_time_array (struct GMT_CTRL *GMT, double min, double max, double inc, char unit, bool interval, double **array) {
```
Comment (joa-quim) [inline src/gmt_support.c]: Iden, misses gmt prefix

**Comment by PaulWessel** [comment]
Right, because you wrap the C I think.  It would probably be a good idea for you/us to maintain a list of lower-level functions etc that you actually use.  It would potentially head off a problem.

**Comment by PaulWessel** [comment]
I am going to commit this guy.  OK @joa-quim ?

File: src/gshhg_version.h [hunk 4]
```diff
@@ -52,7 +52,6 @@ struct GSHHG_VERSION {
 };
 
 /* Prototypes */
-EXTERN_MSC int gshhg_get_version (const char* filename, struct GSHHG_VERSION *gshhg_version);
```
Comment (joa-quim) [inline src/gshhg_version.h]: What if someone whats to know the GSHHG version?

**Comment by joa-quim** [comment]
All functions and structs currently used in Julia are declared in

https://github.com/GenericMappingTools/GMT.jl/blob/master/src/libgmt.jl
https://github.com/GenericMappingTools/GMT.jl/blob/master/src/libgmt_h.jl

Except GMTAPI_DATA_OBJECT that is not actually used.

File: src/gshhg_version.h (hunk 4 above)
Comment (PaulWessel) [reply src/gshhg_version.h]: They can, this is a sub-function of the gshhg_version function.

File: src/gmt_remote.c (hunk 1 above)
Comment (PaulWessel) [reply src/gmt_remote.c]: See #2996 for this discussion - you are both right and wrong on these comments; I will resolve them for now.

**Pull request merged**